src/clinvar_raw/stream.clj

```diff
@@ -248,59 +250,71 @@
+           (filter-by-field (-> order-entry :filter :field)
+                            (-> order-entry :filter :value))
+           (map #(line-map-to-event %
+                                    (:type order-entry)
+                                    release-date
+                                    event-type))))))
+
+(defn process-files [storage-protocol release-date files-to-process]
+  (when (seq files-to-process)
+    (lazy-cat (process-file storage-protocol release-date (first files-to-process))
+              (process-files storage-protocol release-date (rest files-to-process)))))
+
 (defn generate-messages-from-diff
   "Takes a diff notification message, and returns a lazy seq of
    all the output messages in order for this diffed release."
   [parsed-diff-files-msg storage-protocol]
-  (log/debug :fn :generate-messages-from-diff
+  (log/info :fn :generate-messages-from-diff
              :parsed-diff-files-message parsed-diff-files-msg
              :storage-protocol storage-protocol)
   (let [release-date (:release_date parsed-diff-files-msg)]
-    (letfn [(process-file [{:keys [bucket path order-entry event-type]}]
-              (let [reader-fn (partial construct-reader
-                                       storage-protocol
-                                       bucket
-                                       path)]
-                (->> (lazy-line-reader-threaded reader-fn)
-                     (map #(json/parse-string % true))
-                     (filter-by-field (-> order-entry :filter :field)
-                                      (-> order-entry :filter :value))
-                     (map #(line-map-to-event %
-                                              (:type order-entry)
-                                              release-date
-                                              event-type)))))
-            (process-files [files-to-process]
-              (when (seq files-to-process)
-                (lazy-cat (process-file (first files-to-process))
-                          (process-files (rest files-to-process)))))]
-      (->> (for [procedure event-procedures]
-             (let [bucket (:bucket parsed-diff-files-msg)
-                   files (filter-files (:filter-string procedure) (:files parsed-diff-files-msg))]
-               (for [order-entry (:order procedure)
-                     file-path (filter-files (:type order-entry) files)]
-                 {:bucket bucket
-                  :path file-path
-                  :order-entry order-entry
-                  :event-type (:event-type procedure)})))
-           flatten-one-level
-           process-files))))
+    (->> (for [procedure event-procedures]
+           (let [bucket (:bucket parsed-diff-files-msg)
+                 file-list (or (:files parsed-diff-files-msg)
+                               (->> (str (:release_directory parsed-diff-files-msg) "/")
+                                    (gcs/list-all-files bucket)))
```

Comment:
`gcs/list-all-files` should be fine because the `filter-files` will exclude any that don't match the standard path patterns we are expecting. So if there are other, non-release files in the directory those will be ignored as long as they are not within the create/update/delete subdirectories.